src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2165,14 +2174,20 @@ export class StudyViewPageStore
         // We do not support studyIds in the query filters
         let filters: Partial<StudyViewFilter> = {};
         if (query.filterJson) {
-            try {
-                filters = JSON.parse(
-                    decodeURIComponent(query.filterJson)
-                ) as Partial<StudyViewFilter>;
-                this.updateStoreByFilters(filters);
-            } catch (e) {
-                //  TODO: add some logging here?
+            const parsedFilterJson = this.parseRawFilterJson(query.filterJson);
+            if (query.filterJson.includes('patientIdentifiers')) {
+                const samples = await this.fetchSamplesWithStudyIds(studyIds);
```

Comment:
hmmm. it's a little painful to pull down all the samples in this way.  there ought to be an endpoint to retrieve samples for a set of patients.  from what i can see right now, you can only do it for one patient at a time.  @inodb do you think that this is enough of an obscure usage that we shouldn't worry about this?  @haynescd at the very least, we should use the https://genie.cbioportal.org/api/samples/fetch service because that's a lot more likely to be in cache

Reply:
that's gonna result in a shitload of api requests though.  because samples endpoint is likely to be in cache, it's better to just fetch all.

Reply:
Agreed. That is why I initially used the api to get all the samples in a study

Reply:
So you are suggesting to use /studies/{studyId}/samples because it might be cached?

Reply:
yes, i'm pretty sure that the filtering endpoints are not well "covered" by caching for reasons i'm happy to explain.  whereas the samples endpoint IS well covered.